Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,893 @@
+                $providerCount = 0
+                foreach ($provider in $AMSIProviders) {
+                    $provider -match '[0-9A-Fa-f\-]{36}' | Out-Null
+                    foreach ($m in $Matches.Values) {
+                        $key = "HKLM:\SOFTWARE\Classes\ClSid\{$m}"
+                        $providers = $null
+                        $providers = Get-ChildItem $key -ErrorAction SilentlyContinue
+                        if ($providers) {
+                            $providerCount++
+                            Write-Host ""
+                            Write-Host "Provider $($providerCount):"
+                            $providers | Format-Table -AutoSize | Out-Host
+                            $path = $null
+                            $path = ($providers | Where-Object { $_.PSChildName -eq 'InprocServer32' }).GetValue('')
+                            if ($path) {
+                                $WindowsDefenderPath = $path.Substring(1, $path.LastIndexOf("\"))
+                                if ($WindowsDefenderPath -like '*Windows Defender*') {
+                                    Write-Host "Windows Defender with AMSI integration found." -ForegroundColor Green
+                                    $checkCmdLet = $null
+                                    $checkCmdLet = Get-Command Get-MpComputerStatus -ErrorAction SilentlyContinue
+                                    if ($null -eq $checkCmdLet) {
+                                        Write-Warning "Get-MpComputerStatus cmdLet is not available"
+                                    } else {
+                                        if ((Get-MpComputerStatus).RealTimeProtectionEnabled) {
+                                            Write-Host "Windows Defender has Real Time Protection Enabled" -ForegroundColor Green
+                                        } else {
+                                            Write-Warning "Windows Defender has Real Time Protection Disabled"
+                                        }
+                                    }
+                                    Write-Host "It should be version 1.1.18300.4 or newest."
+                                    if (Test-Path $WindowsDefenderPath -PathType Container) {
+                                        $folder = Get-ChildItem  $WindowsDefenderPath | Sort-Object LastWriteTime -Descending | Select-Object -First 1
+                                        $process = Join-Path $folder.FullName "MpCmdRun.exe"
+                                        if (Test-Path $process -PathType Leaf) {
+                                            $DefenderVersion = $null
+                                            $DefenderVersion = [System.Version]::new((& $process -SignatureUpdate | Where-Object { $_.StartsWith('Engine Version:') }).Split(' ')[2])
+                                            if ($DefenderVersion) {
+                                                if (($DefenderVersion.Major -gt 1) -or
```

Comment:
As `$DefenderVersion` is a `[System.Version]` type object, we could simplify the check by just doing an:

`$DefenderVersion -ge "1.1.18300.4"`